src/tests/interactive/cleanup.py

```diff
@@ -40,13 +41,19 @@ def cleanup_one(line):
 
 def main():
     # Writing to stdout converts newlines, which confuses diff on Windows, so
-    # write to a file instead.  There's no reasonable way to do this in a Python
-    # 2/3 compatible way, so the following is Python-2 only.
-    lines = [line.decode("utf-8") for line in sys.stdin]
-    with open(sys.argv[1], mode="wb") as out:
-        for line in lines:
-            out.write(cleanup_one(line).encode("utf-8"))
-            out.flush()
+    # write to a file instead.
+    if sys.version_info < (3,):
+        lines = [line.decode("utf-8") for line in sys.stdin]
+        with open(sys.argv[1], mode="wb") as out:
+            for line in lines:
+                out.write(cleanup_one(line).encode("utf-8"))
+                out.flush()
+    else:
+        lines = [line for line in io.TextIOWrapper(sys.stdin.buffer, encoding="utf-8")]
```

Comment:
I put that in based on https://docs.python.org/3/library/sys.html for stdin/stdout/stderr:

"The character encoding is platform-dependent. Under Windows, if the stream is interactive (that is, if its isatty() method returns True), the console codepage is used, otherwise the ANSI code page. Under other platforms, the locale encoding is used (see locale.getpreferredencoding())."

Reply:
Ouch. Thanks a lot.